src/chat.rs

```diff
@@ -1695,6 +1701,19 @@ pub async fn get_chat_msgs(
                     last_day = curr_day;
                 }
             }
+            if (flags & DC_GCM_ADD_AUTODELETE_MARKER) != 0
+                && autodelete_timer != last_autodelete_timer
+            {
+                ret.push(MsgId::new(DC_MSG_ID_AUTODELETE_MARKER));
+                // Insert a timer in place of msgid.  This way library
+                // user does not need to do additional queries to the
+                // database to find out which timer value the message
+                // has. It prevents race conditions, because the
+                // message may already be deleted by the time its
+                // timer is queried.
+                ret.push(MsgId::new(autodelete_timer));
```

Comment:
i think, adding something else as an ID to the array of messages-IDs might become tricky - the array cannot be accessed easily with that change, cmp. recent discussion on irc.

Reply:
reverted these changes for now, going to use info messages until better markers are implemented